katsdpimageutils/primary_beam_correction.py

```diff
@@ -168,6 +168,8 @@ def inverse_variance(data):
     if len(data) == 0:
         return 0.0
     med, sd = standard_deviation(data)
+    if ~np.isfinite(sd):
+        return 0.0
```

Comment:
I think it might be better to put this a few lines up with the test for `data != 0.0`. Something like: `data = data[(data != 0.0) & (np.isfinite(data))]`.

This would avoid the `nanmedian` in `standard_deviation` checking an array of NaNs and printing a warning about an All-NaN slice.